core/src/main/java/com/radixdlt/p2p/transport/handshake/AuthHandshaker.java

```diff
@@ -181,10 +186,18 @@ public Pair<byte[], AuthHandshakeResult> handleInitialMessage(ByteBuf data) {
       final var encryptedPayload = new byte[data.readableBytes() - sizeBytes.length];
       data.getBytes(sizeBytes.length, encryptedPayload, 0, encryptedPayload.length);
       final var plaintext = ecKeyOps.eciesDecrypt(encryptedPayload, sizeBytes);
-      final var message = serialization.fromDson(plaintext, AuthInitiateMessage.class);
+      final AuthInitiateMessage message;
+      try {
+        message = serialization.fromDson(plaintext, AuthInitiateMessage.class);
+      } catch (DeserializeException ex) {
+        return Pair.of(
+            new byte[] {STATUS_ERROR},
+            AuthHandshakeResult.error(
+                "Auth handshake failed (possible network version mismatch)", Optional.empty()));
```

Comment:
If an Olympia node keeps trying to connect, will we get spammed logs? Do we block them based on IP?

Reply:
We shouldn't be getting spammed logs, the address (on the initiator side) should get blacklisted after a failed handshake